components/aws/sagemaker/tests/integration_tests/component_tests/test_batch_transform_component.py

```diff
@@ -82,4 +83,9 @@ def test_transform_job(
     )
     assert s3_utils.check_object_exists(s3_client, s3_data_bucket, file_key)
 
+    error_message = 'Error in fetching CloudWatch logs for SageMaker job'
+    if argo_utils.find_in_logs(workflow_json["metadata"]["name"], error_message):
```

Comment:
Shouldn't the assert just be based off of this statement instead of the if condition, any particular reason for doing it this way ?

Reply:
Doing it this way might be harder to debug. pytest output shows the assertion that failed and only a little bit of context. This way would show `Test failed: Assertion False` or something along those lines. I'm not sure how easy it would be to determine where this assertion comes from without tracking down the correct line of stdout.

Reply:
This might still be already if you update the print in the next line (L88) with a more descriptive output. Perhaps `Error message was found in Argo logs: {error_message}`